### PR TITLE
Frontend/redirect to created dataset#525

### PR DESCRIPTION
**Pull request by Vatika1**
This branch displays a success message when dataset has been created successfully and redirects the user to site/dataset/:datasetId page on submit.

https://user-images.githubusercontent.com/35615793/113487919-c2d2f380-9488-11eb-9a16-65cb9f20cf51.mp4

### CODE REVIEW

**Comment by ds161833** [comment]
the user should be redirected to the page of the "created" dataset, when the dataset is submitted backend returns the created dataset id, you should redirect the user to /dataset/returnedDatasetId
right now you redirect them to /dataset/undefined 
image

File: Client/src/Components/DatasetUpload/DatasetView.tsx [hunk 1]
```diff
@@ -68,7 +71,11 @@ export const DatasetView = (props: IProps) => {
   }, [])
 
   const handleSubmitForm = async (formDataset: IDatasetModel) => {
-    await callSaveDataset(formDataset)
+    const result = await callSaveDataset(formDataset)
+    if (result > 0) {
+      SnackbarUtils.success("Dataset successfully uploaded")
+    }
+    history.push('/dataset/:' + formDataset.id)
```
Comment (lposo) [inline Client/src/Components/DatasetUpload/DatasetView.tsx]: This part does not redirect to the newly created dataset. formDataset.id is undefined. You should pass the result as the id like so:
const handleSubmitForm = async (formDataset: IDatasetModel) => {
    const result = await callSaveDataset(formDataset)
    if (result > 0) {
      SnackbarUtils.success("Dataset successfully uploaded")
      history.push('/dataset/' + result)
      history.go(0)
    }
  }

**Review by lposo — CHANGES_REQUESTED** [review]

File: Client/src/Components/DatasetUpload/DatasetView.tsx (hunk 1 above)
Comment (Vatika1) [reply Client/src/Components/DatasetUpload/DatasetView.tsx]: done

File: Client/src/Components/DatasetUpload/DatasetForm/DatasetForm.tsx [hunk 2]
```diff
@@ -44,9 +44,10 @@ export const DatasetForm = (props: IProps): any => {
     callListMaterials()
   }, [])
 
-  const handleSubmit = (values: DatasetUploadFormValues) => {
+  const handleSubmit = (values: DatasetUploadFormValues, { resetForm }) => {
     const dataset: IDatasetModel = { ...values.meta, reference: values.reference, data: values.data }
     onSubmit(dataset)
+    //resetForm({ values: '' })
```
Comment (Mehrdadmaskull) [inline Client/src/Components/DatasetUpload/DatasetForm/DatasetForm.tsx]: If this code is not required, then remove it

File: Client/src/Components/DatasetUpload/DatasetView.tsx [hunk 3]
```diff
@@ -68,7 +71,12 @@ export const DatasetView = (props: IProps) => {
   }, [])
 
   const handleSubmitForm = async (formDataset: IDatasetModel) => {
-    await callSaveDataset(formDataset)
+    const result = await callSaveDataset(formDataset)
+    if (result > 0) {
+      SnackbarUtils.success("Dataset successfully uploaded")
+      history.push('/dataset/' + result)
+      //history.go(0)
```
Comment (Mehrdadmaskull) [inline Client/src/Components/DatasetUpload/DatasetView.tsx]: Same with this line

File: Client/src/Components/DatasetUpload/DatasetForm/DatasetForm.tsx (hunk 2 above)
Comment (Vatika1) [reply Client/src/Components/DatasetUpload/DatasetForm/DatasetForm.tsx]: it was removed in the latest push @Mehrdadmaskull

File: Client/src/Components/DatasetUpload/DatasetView.tsx (hunk 3 above)
Comment (Vatika1) [reply Client/src/Components/DatasetUpload/DatasetView.tsx]: it was removed in the latest push @Mehrdadmaskull

**Review by lposo — APPROVED** [review]
Looks good.

**Review by Mehrdadmaskull — APPROVED** [review]
LGTM

**Pull request merged**